controller/controller.go

```diff
@@ -84,6 +84,10 @@ type Impl struct {
 	// simultaneously in two different workers.
 	WorkQueue workqueue.RateLimitingInterface
 
+	// entryTimes keeps the entry time of a given item in the
+	// WorkQueue, which is used on removal to calculate time spent queuing
```

Comment:
The `sync.Map` docs suggest that it should be used in some specific cases (this isn't that case) and instead recommends to use mutex+map.

Reply:
will change